blocks/core/ff_module/ff_module-task-event-repeater/ff_module-task-event-repeater.js

```diff
@@ -14,26 +15,29 @@ module.exports = React.createClass({
     render: function(){
         return <div className="ff_container-task-event-repeater">
             <ol className="ff_container-task-event-repeater__items">
-                {this.getEvents().map(event=>{
+                {this.getGroups().map(events=>{
                     return  <li className="ff_container-task-event-repeater__item"
-                                key={'event-'+event.localEventId}>
-                                <TaskEvent {...event}/>
+                                key={this.getKey(events)}>
+                                <TaskEventGroup events={events}/>
                             </li>
                 })}
             </ol>
         </div>
     },
     propTypes: {
-        events: React.PropTypes.arrayOf(React.PropTypes.shape(TaskEvent.PropTypes)).isRequired
+        eventGroups: React.PropTypes.arrayOf(React.PropTypes.arrayOf(React.PropTypes.shape(TaskEvent.PropTypes))).isRequired
     },
-    getEvents: function() {
-        var events = this.props.events;
-        if (isArray(events)) {
-            events = events.sort(function(a, b) {
-                // Reverse chronological order
-                return ensureIsDate(b.description.sent) - ensureIsDate(a.description.sent);
-            });
-        }
-        return events;
+    getKey:function(events){
+        return events.reduce((memo, event)=> '' + memo + event.localEventId, 'group-');
+    },
+    getGroups: function() {
+        var eventGroups = this.props.eventGroups || [];
+        // if (isArray(events)) {
+        //     events = events.sort(function(a, b) {
+        //         // Reverse chronological order
+        //         return ensureIsDate(b.description.sent) - ensureIsDate(a.description.sent);
+        //     });
+        // }
```

Comment:
Don't we still need to apply some sorting? Both on the events in a group and the groups themselves?